include/yamail/resource_pool/async/detail/pool_impl.hpp

```diff
@@ -95,36 +97,27 @@ class pool_impl {
     using value_type = Value;
     using io_context_t = IoContext;
     using idle = resource_pool::detail::idle<value_type>;
-    using list = std::list<idle>;
-    using list_iterator = typename list::iterator;
+    using storage_type = resource_pool::detail::storage<value_type>;
+    using list_iterator = typename storage_type::cell_type;
```

Comment:
I'd suggest keeping `*_iterator` in the name of the type that represents iterator.

Reply:
Do you mean this?
```c++
using cell_type = typename storage_type::cell_type;
```

Reply:
No, I meant 

```c++
using list_iterator = typename storage_type::cell_iterator;
```

because this is an iterator, not a cell.